Sources/FoundationEssentials/String/String+Path.swift

```diff
@@ -512,28 +468,81 @@ extension String {
         }
         
         #if !os(WASI) // WASI does not have user concept
-        // Next, attempt to find the home directory via getpwnam/getpwuid
-        if let user {
-            if let dir = Platform.homeDirectory(forUserName: user) {
-                return dir.standardizingPath
-            }
-        } else {
-            // We use the real UID instead of the EUID here when the EUID is the root user (i.e. a process has called seteuid(0))
-            // In this instance, we historically do this to ensure a stable home directory location for processes that call seteuid(0)
-            if let dir = Platform.homeDirectory(forUID: Platform.getUGIDs(allowEffectiveRootUID: false).uid) {
-                return dir.standardizingPath
-            }
+        // Next, attempt to find the home directory via getpwuid
+        // We use the real UID instead of the EUID here when the EUID is the root user (i.e. a process has called seteuid(0))
+        // In this instance, we historically do this to ensure a stable home directory location for processes that call seteuid(0)
+        if let dir = Platform.homeDirectory(forUID: Platform.getUGIDs(allowEffectiveRootUID: false).uid) {
+            return dir.standardizingPath
         }
-        #endif // !os(WASI)
-
+        #endif
+        
         // Fallback to HOME for the current user if possible
-        if user == nil, let home = getenv("HOME") {
+        if let home = getenv("HOME") {
             return String(cString: home).standardizingPath
         }
         
-        // If all else fails, log and fall back to /var/empty
+        // If all else fails, fall back to /var/empty
         return "/var/empty"
-        #endif // os(Windows)
+        #endif
+    }
+
+    internal static func homeDirectoryPath(forUser user: String) -> String? {
+        #if os(Windows)
+        guard !user.isEmpty else {
+            return nil
```

Comment:
Should we return `nil` for empty user on non-Windows platforms, too?

Reply:
Oh, I see that `Platform.homeDirectory(forUserName: "")` returns `nil`.

Reply:
Yeah we have some unit tests that check for this behavior that aren't platform specific so in the end I think the behavior matches, but happy to restructure if you think this should be moved out of the `#if` block